app/src/main/java/org/piwigo/ui/main/MainActivity.java

```diff
@@ -183,50 +185,47 @@ public void onPropertyChanged(Observable sender, int propertyId) {
+                if (viewModel.loginStatus.get() == STAT_STATUS_FETCHED) {
+
+                    Fragment f = getSupportFragmentManager().findFragmentById(R.id.content);
+                    Log.d("MainActivity", "fragment " + f.toString());
+                    if (f instanceof AlbumsFragment) {
+                        Integer cat = ((AlbumsFragment) f).getViewModel().getCategory();
+                        Log.d("MainActivity", "category " + cat);
+                        if (cat != null) {
+                            ((AlbumsFragment) f).getViewModel().loadAlbums(cat);
+                        } else {
+                            ((AlbumsFragment) f).getViewModel().loadAlbums(0);
+                        }
+                        ((AlbumsFragment) f).getViewModel().onRefresh();
+
+                    }
+                }
+            }
+        });
+
         final Observer<Account> accountObserver = account -> {
+            Log.d("MainActivity", "accounts changed " + currentAccount.toString());
             // reload the albums on account changes
             if (account != null && !account.equals(currentAccount)) {
                 currentAccount = account;
-                viewModel.username.set(userManager.getUsername(account));
-                viewModel.url.set(userManager.getSiteUrl(account));
-                viewModel.displayFab.set(!userManager.isGuest(currentAccount));
-                /* Login to the new site after account changes.
-                 * It seems quite unclean to do that here -> TODO: FIXME*/
-                rx.Observable<LoginResponse> a = userRepository.login(account);
-                a.subscribe(new rx.Observer<LoginResponse>() {
-                    @Override
-                    public void onCompleted() {
-                    }
-
-                    @Override
-                    public void onError(Throwable e) {
-                        Log.e(TAG, "Login failed: " + e.getMessage());
-                        // TODO: notify loginfailure
-                    }
-
-                    @Override
-                    public void onNext(LoginResponse loginResponse) {
-                        Log.i(TAG, "Login succeeded: " + loginResponse.pwgId + " token: " + loginResponse.statusResponse.result.pwgToken);
-                        userManager.setCookie(account, loginResponse.pwgId);
-                        userManager.setToken(account, loginResponse.statusResponse.result.pwgToken);
-                        userManager.setChunkSize(account, loginResponse.statusResponse.result.uploadFormChunkSize);
-                    }
-                });
-                initStartFragment(viewModel);
+                viewModel.changeAccount(account);
             }
             if (account == null) {
                 viewModel.username.set("");
                 viewModel.url.set("");
             }
         };
+        initStartFragment();
```

Comment:
This change I don't catch. Don't you reset the selected album with this to the root on device rotation?

Reply:
Honestly, this is the part that needs most work. But I better revert here.

The problem was that `initStartFragment()` loaded the root gallery before the login was finished, so I removed the call - but I had to readd it because it's also needed to actually show pictures :)